cw-orch/src/interchain/follow_ibc_execution.rs

```diff
@@ -0,0 +1,211 @@
+///         This is also logged for debugging purposes
+///
+/// 4. Finally, some additionnal packets may have been sent out during the whole process. We need to check the generated transactions for additional packets.
+///    This 4th step make the whole process recursive. Those two transactions potentially sends out additional packets !
+///         - The receive Packet transaction identified in 2. on the distant chain
+///         - The receive Acknowledgement transaction identified in 3.b. on the origin chain
+
+#[async_recursion::async_recursion]
+pub async fn follow_trail(chain1: String, channel1: Channel, tx_hash: String) -> Result<()> {
+    // 1. Getting IBC related events for the current tx
+    let tx = Node::new(channel1.clone()).find_tx(tx_hash.clone()).await?;
+
+    let send_packet_events = tx.get_events("send_packet");
+    if send_packet_events.is_empty() {
+        return Ok(());
+    }
+
+    log::info!(
+        target: &chain1,
+        "Investigating sent packet events on tx {}",
+        tx_hash
+    );
+    let connections = get_events(&send_packet_events, "packet_connection");
+    let src_ports = get_events(&send_packet_events, "packet_src_port");
+    let src_channels = get_events(&send_packet_events, "packet_src_channel");
+    let sequences = get_events(&send_packet_events, "packet_sequence");
+    let packet_datas = get_events(&send_packet_events, "packet_data");
+    let chain_ids = try_join_all(
+        connections
+            .iter()
+            .map(|c| async {
+                Ok::<_, anyhow::Error>(
+                    Ibc::new(channel1.clone())
+                        .connection_client(c.clone())
+                        .await?
+                        .chain_id,
+                )
+            })
+            .collect::<Vec<_>>(),
+    )
+    .await?;
+
+    // We log the packets we follow.
+    for i in 0..src_ports.len() {
+        log::info!(
+            target: &chain1,
+            "IBC packet n° {}, sent on {} on tx {}, with data: {}",
+            sequences[i],
+            chain1,
+            tx_hash,
+            packet_datas[i]
+        );
+    }
+
+    // We follow the IBC trail for all packets found inside the transaction
+    let txs_to_follow = try_join_all(
+        src_ports
+            .iter()
+            .enumerate()
+            .map(|(i, _)| {
+                let chain_a = chain1.clone();
+                let port_a = src_ports[i].clone();
+                let channel_a = src_channels[i].clone();
+
+                let chain_b = chain_ids[i].clone();
+
+                let sequence = sequences[i].clone();
+
+                async move {
+                    // That's all we need to generate an InterchainChannel object.
+                    let interchain_channel = InterchainChannelBuilder::default()
+                        .chain_a(chain_a.clone())
+                        .port_a(port_a)
+                        .is_local_chain_a()
```

Comment:
what are these is_local functions?

Reply:
To indicate what grpc type we want, we need to discuss